src/core/algorithms/dd/split/split.cpp

```diff
@@ -665,29 +705,21 @@ std::list<DD> Split::InstanceExclusionReduce(
 
     if (!last_dd_holds) {
         std::vector<DF> remainder = DoNegativePruning(search, last_df);
-        return InstanceExclusionReduce(tuple_pairs, remainder, rhs, cnt);
+        return InstanceExclusionReduce(tuple_pair_indices, remainder, rhs, cnt);
     }
 
     auto const [prune, remainder] = PositiveSplit(search, first_df);
 
-    dds = InstanceExclusionReduce(tuple_pairs, remainder, rhs, cnt);
+    dds = InstanceExclusionReduce(tuple_pair_indices, remainder, rhs, cnt);
     std::list<DD> const pruning_dds =
-            InstanceExclusionReduce(remaining_tuple_pairs, prune, rhs, cnt);
+            InstanceExclusionReduce(remaining_tuple_pair_indices, prune, rhs, cnt);
 
     std::list<DD> merged_dds = MergeReducedResults(dds, pruning_dds);
     dds.splice(dds.end(), merged_dds);
```

Comment:
again, I think this should be done:
```suggestion
    dds.splice(dds.end(), std::move(merged_dds));
```